hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/launcher/ContainerLaunch.java

```diff
@@ -270,7 +270,7 @@ public Integer call() {
         
         // Write out the environment
         exec.writeLaunchEnv(containerScriptOutStream, environment, localResources,
-            launchContext.getCommands());
+            launchContext.getCommands(), new Path(containerLogDirs.get(0)));
```

Comment:
Why is there more than 1? Could the list be empty?

Reply:
I'm trusting the initial patch here...but yes it should not be of size 0 otherwise it means the container cannot log anywhere. Things would have been stopped before this occurs.